app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,25 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    if @post.save
+      redirect_to user_path(id: @post.author_id)
+    else
+      render :new, alert: 'An error has occurred while creating the post'
+    end
+  end
```

Comment:
✅ A post is created on behalf of the currently logged in user (current_user) of the app 👍🏻 

Kindly refactor this to 
```ruby
def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```